packages/openspp/src/Adaptor.js

```diff
@@ -660,6 +658,311 @@ export function unenroll(registrant_id, program_id) {
+          'g2p.group.membership.kind',
+          [['name', '=', role]]
+        );
+        if (roleId.length === 0) {
+          roleId = [await sppConnector.create(
+            'g2p.group.membership.kind',
+            { name: role }
+          )];
+        }
+      }
+      let res = await sppConnector.searchRead(
+        'g2p.group.membership',
+        [
+          ['group.registrant_id', '=', group_id],
+          ['individual.registrant_id', '=', individual_id],
+          ['is_ended', '=', false],
+        ],
+        ['id', 'kind'],
+        { limit: 1 },
+      );
+      if (res.length === 0) {
+        let individual = await sppConnector.searchRead(
+          'res.partner',
+          [
+            ['registrant_id', '=', individual_id],
+            ['is_registrant', '=', true],
+            ['is_group', '=', false],
+          ],
+          ['id'],
+          { limit: 1 },
+        );
+        let group = await sppConnector.searchRead(
+          'res.partner',
+          [
+            ['registrant_id', '=', group_id],
+            ['is_registrant', '=', true],
+            ['is_group', '=', true],
+          ],
+          ['id'],
+          { limit: 1 },
+        );
+        if (individual.length === 0 || group.length === 0) {
+          throw new Error(`Individual or Group does not exist!`);
+        }
+        await sppConnector.create(
+          'g2p.group.membership',
+          {
+            individual: individual[0].id,
+            group: group[0].id,
+            kind: [[6, 0, roleId]],
+          }
+        );
+      } else {
+        let groupMembershipIds = res.map( i => i.id );
+        await sppConnector.update(
+          'g2p.group.membership',
+          groupMembershipIds,
+          { kind: [[6, 0, roleId]] },
+        );
+      }
+      if (role.length > 0) {
+        console.log(`ℹ Individual ${individual_id} added to group ${group_id} with role ${role}!`);
+      } else {
+        console.log(`ℹ Individual ${individual_id} added to group ${group_id}!`);
+      }
+    } catch (err) {
+      console.log(`✗ Error: ${err}`);
+    } finally {
+      return state;
+    }
+  }
+}
+
+/**
+ * remove individual from group in OpenSPP
+ * @public
+ * @example
+ * removeFromGroup("GRP_B2BRHJN2", "IND_8DUQL4M4")
+ * @function
+ * @param {string} group_id - group registrant id
+ * @param {string} individual_id - individual registrant id
+ * @returns {Operation}
+ */
+export function removeFromGroup(group_id, individual_id) {
+  return async state => {
+    try {
+      let res = await sppConnector.searchRead(
+        'g2p.group.membership',
+        [
+          ['group.registrant_id', '=', group_id],
+          ['individual.registrant_id', '=', individual_id],
+          ['is_ended', '=', false],
+        ],
+        ['id'],
+      );
+      if (res.length > 0) {
+        let groupMembershipIds = res.map( i => i.id );
+        let now = new Date();
+        let sppDateTimeNowString =
```

Comment:
I am reluctant to suggest this because I think it's broken in the CLI and it's not really a pattern I'm keen to support... but the `common` adaptor exports a `dateFns` objects. dateFns has a `format` function which is really useful for this stuff: https://date-fns.org/v2.16.1/docs/format

If it works it'll let you do something like this:
```
 let sppDateTimeNowString = format (new Date(), 'y-M-d H:m:s')`
```

Reply:
done replacing